File: src/main/java/ar/edu/um/tesoreria/rest/model/CategoriaContratado.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class CategoriaContratado extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = -7627833487679606596L;

	@Id
	private Integer categoriaId;

	private String nombre = "";
	private BigDecimal importe = BigDecimal.ZERO;

}

File: src/main/java/ar/edu/um/tesoreria/rest/kotlin/model/CategoriaContratado.kt
package ar.edu.um.tesoreria.rest.kotlin.model

import ar.edu.um.tesoreria.rest.model.Auditable
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table
data class CategoriaContratado(

    @Id
    var categoriaId: Int? = null,

    var nombre: String = "",
    var importe: BigDecimal = BigDecimal.ZERO,

    ) : Auditable()
